phylo/src/likelihood/mod.rs

```diff
@@ -1,9 +1,9 @@
 use crate::substitution_models::FreqVector;
 
-pub trait LikelihoodCostFunction<'a, const N: usize> {
+pub trait LikelihoodCostFunction<'a> {
     type Model;
     type Info;
-    fn compute_log_likelihood(&self, model: &Self::Model) -> f64;
+    fn compute_log_likelihood(&self) -> f64;
```

Comment:
nit: In Rust verbs as methods usually mean that this method modifies the state of this object. Since it doesn't do it, it is better to use nouns
https://rust-lang.github.io/api-guidelines/naming.html

This case is a bit funny, because `log_` wouldn't be a good prefix for the method name (-; But instead of returning f64, you would return a `LogarithmicValue`, for instance and suddenly it might make sense to write it as
```rust
fn likelihood(&self) -> LogarithmicValue;
```
or smth like that

Please note that this is a super-duper nitpicking happening here.

Reply:
> In Rust verbs as methods usually mean that this method modifies the state of this object. Since it doesn't do it, it is better to use nouns

I cannot find this convention anywhere, neither at the provided link nor online. I would love to see it; it makes a lot of sense.

That said, the method name is, in fact, garbage (: It will be refactored in the next PR, so will leave it as is here.